cdap-api/src/main/java/co/cask/cdap/api/flow/flowlet/AbstractFlowlet.java

```diff
@@ -106,6 +110,70 @@ protected void setProperties(Map<String, String> properties) {
   }
 
   /**
+   * @see FlowletConfigurer#addStream(Stream)
+   */
+  protected final void addStream(Stream stream) {
```

Comment:
@albertshau For the second question, the PR now allows Flows to add streams/datasets.